utils/release/git-lock.mjs

```diff
@@ -42,10 +46,15 @@ const ensureUpToDateBranch = async () => {
  * This will make .github\workflows\git-lock-fail.yml run and thus fail the associated check.
  */
 const lockBranch = async () => {
+  const DEPLOY_KEY = process.env.DEPLOY_KEY;
+  if (DEPLOY_KEY === undefined) {
+    throw new Error('Deploy key is undefined');
+  }
+  await gitSetupSshRemote(REPO_OWNER, REPO_NAME, DEPLOY_KEY, GIT_SSH_REMOTE);
```

Comment:
I'm worried that updating the remote here may have unintended side-effects in workflows that call `lockBranch`, we should probably move `gitSetupSshRemote` out of here.